### ISSUE TITLE
Fixed route failure on machine

### ISSUE THREAD

**Issue by rowleya** [SpiNNakerManchester/PACMAN#172]
The following demonstrates a failure of the the FixedRouteRouter.  Although this uses a virtual machine, this machine was a real machine that was read quite a long time after it had been booted.  It still could be an "odd" machine, but this needs to be investigated.

```
from spinn_machine.virtual_machine import VirtualMachine
from pacman.model.placements.placements import Placements
from pacman.model.placements.placement import Placement
from pacman.operations.fixed_route_router.fixed_route_router \
    import FixedRouteRouter
import logging

logging.basicConfig(level=logging.INFO)


class DestinationVertex(object):
    pass


down_chips = {(9, 4), (12, 4), (12, 7), (13, 8)}
down_cores = {(0, 0, 1), (0, 1, 1), (0, 2, 1), (0, 3, 1), (1, 0, 1), (1, 1, 1), (1, 1, 17), (1, 2, 1), (1, 3, 1), (1, 3, 17), (1, 4, 1), (2, 0, 1), (2, 1, 1), (2, 2, 1), (2, 2, 17), (2, 3, 1), (2, 4, 1), (2, 5, 1), (3, 0, 1), (3, 1, 1), (3, 1, 17), (3, 2, 1), (3, 3, 1), (3, 3, 16), (3, 3, 17), (3, 4, 1), (3, 5, 1), (3, 5, 17), (3, 6, 1), (4, 0, 1), (4, 1, 1), (4, 2, 1), (4, 3, 1), (4, 4, 1), (4, 5, 1), (4, 6, 1), (4, 7, 1), (4, 7, 17), (4, 8, 1), (4, 9, 1), (4, 10, 1), (4, 11, 1), (5, 1, 1), (5, 2, 1), (5, 2, 17), (5, 3, 1), (5, 4, 1), (5, 4, 17), (5, 5, 1), (5, 6, 1), (5, 6, 17), (5, 7, 1), (5, 8, 1), (5, 9, 1), (5, 9, 17), (5, 10, 1), (5, 11, 1), (5, 11, 17), (5, 12, 1), (6, 2, 1), (6, 3, 1), (6, 4, 1), (6, 5, 1), (6, 6, 1), (6, 7, 1), (6, 8, 1), (6, 9, 1), (6, 10, 1), (6, 11, 1), (6, 12, 1), (6, 13, 1), (7, 3, 1), (7, 4, 1), (7, 5, 1), (7, 6, 1), (7, 7, 1), (7, 8, 1), (7, 9, 1), (7, 9, 17), (7, 10, 1), (7, 11, 1), (7, 11, 17), (7, 12, 1), (7, 13, 1), (7, 13, 17), (7, 14, 1), (8, 4, 1), (8, 5, 1), (8, 6, 1), (8, 7, 1), (8, 7, 16), (8, 7, 17), (8, 8, 1), (8, 9, 1), (8, 10, 1), (8, 11, 1), (8, 12, 1), (8, 13, 1), (8, 14, 1), (8, 15, 1), (9, 5, 1), (9, 5, 17), (9, 6, 1), (9, 7, 1), (9, 7, 16), (9, 7, 17), (9, 8, 1), (9, 9, 1), (9, 10, 1), (9, 10, 17), (9, 11, 1), (9, 12, 1), (9, 12, 17), (9, 13, 1), (9, 14, 1), (9, 14, 17), (9, 15, 1), (10, 4, 1), (10, 4, 17), (10, 5, 1), (10, 6, 1), (10, 6, 16), (10, 6, 17), (10, 7, 1), (10, 8, 1), (10, 9, 1), (10, 9, 16), (10, 9, 17), (10, 10, 1), (10, 11, 1), (10, 12, 1), (10, 13, 1), (10, 14, 1), (10, 15, 1), (11, 4, 1), (11, 5, 1), (11, 5, 17), (11, 6, 1), (11, 6, 17), (11, 7, 1), (11, 7, 17), (11, 8, 1), (11, 9, 1), (11, 9, 17), (11, 10, 1), (11, 11, 1), (11, 12, 1), (11, 13, 1), (11, 14, 1), (11, 15, 1), (12, 0, 1), (12, 1, 1), (12, 2, 1), (12, 3, 1), (12, 5, 1), (12, 6, 1), (12, 6, 17), (12, 8, 1), (12, 8, 17), (12, 9, 1), (12, 9, 16), (12, 9, 17), (12, 10, 1), (12, 11, 1), (12, 11, 17), (13, 0, 1), (13, 1, 1), (13, 1, 16), (13, 1, 17), (13, 2, 1), (13, 3, 1), (13, 3, 17), (13, 4, 1), (13, 4, 17), (13, 5, 1), (13, 5, 17), (13, 6, 1), (13, 6, 17), (13, 7, 1), (13, 9, 1), (13, 9, 17), (13, 10, 1), (13, 10, 17), (13, 11, 1), (14, 0, 1), (14, 1, 1), (14, 2, 1), (14, 3, 1), (14, 4, 1), (14, 4, 17), (14, 5, 1), (14, 6, 1), (14, 6, 17), (14, 7, 1), (14, 7, 17), (14, 8, 1), (14, 8, 17), (14, 9, 1), (14, 10, 1), (14, 11, 1), (14, 11, 16), (14, 11, 17), (15, 0, 1), (15, 1, 1), (15, 1, 17), (15, 2, 1), (15, 3, 1), (15, 3, 17), (15, 4, 1), (15, 5, 1), (15, 5, 17), (15, 6, 1), (15, 7, 1), (15, 8, 1), (15, 9, 1), (15, 10, 1), (15, 11, 1), (16, 0, 1), (16, 1, 1), (16, 2, 1), (16, 2, 17), (16, 3, 1), (16, 4, 1), (16, 5, 1), (16, 6, 1), (16, 7, 1), (16, 8, 1), (16, 9, 1), (16, 10, 1), (16, 11, 1), (17, 1, 1), (17, 2, 1), (17, 2, 17), (17, 3, 1), (17, 4, 1), (17, 4, 17), (17, 5, 1), (17, 6, 1), (17, 6, 17), (17, 7, 1), (17, 8, 1), (17, 9, 1), (17, 9, 17), (17, 10, 1), (17, 11, 1), (17, 11, 17), (17, 12, 1), (18, 2, 1), (18, 3, 1), (18, 4, 1), (18, 5, 1), (18, 6, 1), (18, 7, 1), (18, 8, 1), (18, 9, 1), (18, 10, 1), (18, 11, 1), (18, 12, 1), (18, 13, 1), (19, 3, 1), (19, 4, 1), (19, 5, 1), (19, 6, 1), (19, 7, 1), (19, 7, 17), (19, 8, 1), (19, 9, 1), (19, 10, 1), (19, 11, 1), (19, 11, 17), (19, 12, 1), (19, 13, 1), (19, 13, 17), (19, 14, 1), (20, 4, 1), (20, 5, 1), (20, 6, 1), (20, 7, 1), (20, 8, 1), (20, 9, 1), (20, 10, 1), (20, 11, 1), (20, 12, 1), (20, 13, 1), (20, 14, 1), (20, 15, 1), (21, 4, 1), (21, 5, 1), (21, 5, 17), (21, 6, 1), (21, 7, 1), (21, 7, 17), (21, 8, 1), (21, 9, 1), (21, 10, 1), (21, 10, 17), (21, 11, 1), (21, 12, 1), (21, 12, 17), (21, 13, 1), (21, 14, 1), (21, 14, 17), (21, 15, 1), (22, 4, 1), (22, 5, 1), (22, 6, 1), (22, 7, 1), (22, 7, 17), (22, 8, 1), (22, 9, 1), (22, 10, 1), (22, 11, 1), (22, 12, 1), (22, 13, 1), (22, 14, 1), (22, 15, 1), (23, 4, 1), (23, 5, 1), (23, 5, 17), (23, 6, 1), (23, 7, 1), (23, 7, 17), (23, 8, 1), (23, 9, 1), (23, 9, 17), (23, 10, 1), (23, 11, 1), (23, 12, 1), (23, 13, 1), (23, 14, 1), (23, 15, 1), (24, 4, 1), (24, 5, 1), (24, 6, 1), (24, 7, 1), (24, 8, 1), (24, 9, 1), (24, 10, 1), (24, 11, 1), (25, 5, 1), (25, 6, 1), (25, 6, 17), (25, 7, 1), (25, 8, 1), (25, 8, 17), (25, 9, 1), (25, 10, 1), (25, 10, 17), (25, 11, 1), (25, 11, 17), (26, 6, 1), (26, 7, 1), (26, 8, 1), (26, 9, 1), (26, 10, 1), (26, 11, 1), (27, 7, 1), (27, 8, 1), (27, 9, 1), (27, 10, 1), (27, 11, 1)}
down_links = {(1, 2, 1), (1, 3, 0), (2, 2, 2), (2, 4, 5), (3, 3, 3), (3, 4, 4), (8, 4, 0), (9, 5, 5), (10, 4, 3), (10, 5, 4), (11, 4, 0), (11, 6, 1), (11, 7, 0), (12, 3, 2), (12, 5, 5), (12, 6, 2), (12, 8, 0), (12, 8, 5), (13, 4, 3), (13, 5, 4), (13, 7, 2), (13, 7, 3), (13, 9, 5), (14, 8, 3), (14, 9, 4)}

machine = VirtualMachine(
    width=28, height=16, down_chips=down_chips, down_cores=down_cores,
    down_links=down_links)

placements = Placements()
for chip in machine.ethernet_connected_chips:
    placements.add_placement(
        Placement(DestinationVertex(), chip.x, chip.y, 4))

router = FixedRouteRouter()
router(machine, placements, 5, DestinationVertex)
```

**Comment by rowleya** [comment]
Note that looking at the "down links" above, this could be a machine with some unidirectional links.  I believe that this is a problem for the Dijkstra router, since it relies on using reverse paths through the machine after a forward path has been formed.

**Comment by rowleya** [comment]
Looking into this further, I have tried this again with "RigRoute" as the router, and the error here is that it can't find a route between (2, 3) and (2, 4).  This is correct - all the routes incoming to chip 2, 3 are down!  Fix is probably therefore to mark a chip where all links are down as unusable...

**Comment by rowleya** [comment]
This has been fixed in SpiNNakerManchester/SpiNNMachine#81 and SpiNNakerManchester/SpiNNMan#133.

**Comment by andrewgait** [comment]
Not sure why this issue didn’t get closed a year ago...
